File: src/main/java/com/ascargon/rocketshow/util/ActionSystem.java
package com.ascargon.rocketshow.util;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ActionSystem extends Action {

    public enum SystemActionType {
        REBOOT
    }

    private SystemActionType systemActionType;

    @Override
    public ActionType getType() {
        return ActionType.SYSTEM;
    }
}

File: src/main/webapp/src/app/models/action-system.ts
import { Action } from "./action";

export class ActionSystem extends Action {
  systemActionType: string;

  constructor(data?: any) {
    super(data);

    if (!data) {
      return;
    }

    this.systemActionType = data.systemActionType;

    if (!this.systemActionType) {
      this.systemActionType = "REBOOT";
    }
  }

  toJSON() {
    return { actionSystem: { ...this } };
  }
}
